src/gordon_gcp/clients/gdns.py

```diff
@@ -97,47 +97,109 @@ class GDNSClient(http.AIOConnection):
             object attached to :obj:`auth_client` if not provided.
     """
     BASE_URL = 'https://www.googleapis.com/dns'
+    DNS_CHANGES_DONE = 'done'
 
     def __init__(self, project=None, auth_client=None, api_version='v1',
                  session=None):
         super().__init__(auth_client=auth_client, session=session)
         self.project = project
         self._base_url = f'{self.BASE_URL}/{api_version}/projects/{project}'
 
-    def _parse_resp_to_records(self, response, records):
-        unparsed_records = response.get('rrsets', [])
-        for record in unparsed_records:
-            rrset = GCPResourceRecordSet(**record)
-            records.append(rrset)
+    @staticmethod
+    def get_rrsets_as_objects(rrsets):
+        """Return a list of rrsets as GCPResourceRecordSets.
+
+        Args:
+            rrsets (list of dict): RRsets represented as dicts.
+        Returns:
+            list of :class:`GCPResourceRecordSet` objects.
+        """
+        return [GCPResourceRecordSet(**rrset) for rrset in rrsets]
+
+    def get_managed_zone(self, zone):
+        """Get the GDNS managed zone name for a DNS zone.
+
+        Google uses custom string names with specific `requirements
+        <https://cloud.google.com/dns/api/v1/managedZones#resource>`_
+        for storing records. The scheme implemented here chooses a
+        managed zone name which removes the trailing dot and replaces
+        other dots with dashes, and in the case of reverse records,
+        uses only the two most significant octets, prepended with
+        'reverse'.
+
+        Example:
+           get_managed_zone('example.com.') = 'example-com'
+           get_managed_zone('30.20.10.in-addr.arpa.) = 'reverse-20-10'
+
+        Args:
+            zone (str): DNS zone.
+        Returns:
+            str of managed zone name.
+
+        """
+        if zone.endswith('.in-addr.arpa.'):
+            return 'reverse-' + '-'.join(zone.split('.')[1:3])
+        return '-'.join(zone.split('.')[:-1])
 
-    async def get_records_for_zone(self, zone):
+    async def get_records_for_zone(self, zone, params=None):
         """Get all resource record sets for a particular managed zone.
```

Comment:
Actually I meant that you're actually passing in a DNS zone, so the docstring (and probably the function name) should reflect that.

Reply:
Well, it's a little trickier than that. The records for the relevant DNS zone is requested (which is why that is indicated in args part of the docstring), but what is returned is the records for the managed zone that corresponds to that DNS zone (which is why I left it in this part of the docstring).  Because gordon-gcp currently puts all records in the same DNS zone, this is not an important distinction, *but* if any user of the client were to pass in reverse records it is significant, since the scheme we use for now has a many to one relationship of DNS zone[s] -> managed zone because we reduce to the two most significant octets in reverse zones.  For example `get_records_for_zone('30.20.10.in-addr.arpa')` returns records from that `30.20.10.in-addr.arpa`, but also might return records from `31.20.10.in-addr.arpa`.

Reply:
I would argue that the DNS zone in that case is 20.10.in-addr.arpa.  Just depends where you stick the SOA and NS.  I don't care super much as long as it's clear that it's taking one thing and returning (essentially) another.